File: backend/TravelPlanService/Models/ChecklistItem.cs
namespace TravelPlanService.Models;

public class ChecklistItem
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public Guid TravelPlanId { get; set; }
    public string Name { get; set; } = string.Empty;
    public bool IsCompleted { get; set; }
    public int OrderIndex { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public TravelPlan TravelPlan { get; set; } = null!;
}

File: frontend/src/models/ChecklistItem.js
export class ChecklistItem {
  constructor(data = {}) {
    this.id = data.id || '';
    this.travelPlanId = data.travelPlanId || '';
    this.name = data.name || '';
    this.isCompleted = data.isCompleted || false;
    this.orderIndex = data.orderIndex || 0;
    this.createdAt = data.createdAt || null;
  }
}
